woocommerce/woocommerce-ios
# [Mobile Payments] Return card reader connection tokens from wcpay or stripe endpoints

Closes: #5535

@jaclync @ctarda - just one review needed - but y'all are both welcome of course

### Why?
- Connecting to the Bluetooth card readers requires a token from Stripe. This PR adds the remote to fetch that token from an endpoint hosted on the Stripe extension on the merchant's self-hosted site. This is similar to what we do for the WooCommerce Payments extension.

### Nuances
- I renamed WCPayConnectionToken to  ReaderConnectionToken and have both the WCPayRemote and the StripeRemote loadConnectionToken function return that
- This differs from what I did for WCPayAccount and StripeAccount in #5799 
- My rationale is that the connection token object is super simple - it only has two properties - the token and a test mode boolean, and we only appear to use the token
- This differs from WCPayAccount and StripeAccount which have many more fields and which seem ripe for differing fields/logic/etc

### Description
- Renames `WCPayConnectionToken` to `ReaderConnectionToken` (and `WCPayConnectionTokenMapper` to `ReaderConnectionTokenMapper`)
- Adds `loadConnectionToken` to `StripeRemote` similar to what was done for WCPayRemote
- Adds unit tests

### Testing instructions
- Ensure unit tests pass.
- We don't call this yet (that's coming in a subsequent PR), but if you'd like, for a site with the Stripe extension active and set up (test mode is fine), add the following code to `PaymentGatewayAccountStore` and ensure the breakpoint is hit with success and a token

<img width="1538" alt="Screen Shot 2022-01-07 at 11 01 44 AM" src="https://user-images.githubusercontent.com/1595739/148594252-a57bcb53-692f-4e2c-b222-a7a3869bc110.png">

### Screenshots
N/A

### Related
- https://github.com/woocommerce/woocommerce-gateway-stripe/pull/2129

---
